File: frontend/src/services/orderService.js
// orderService.js

import api from "./api";

export const placeOrder = async (addressId) => {
  const res = await api.post("/orders", {
    addressId,
  });

  return res.data.data;
};

export const getMyOrders = async () => {
  const res = await api.get("/orders");

  return res.data.data;
};


export const getOrderById = async (orderId) => {
  const res = await api.get(`/orders/${orderId}`);

  return res.data.data;
};

export const getAllOrders = async () => {
  const res = await api.get("/orders/admin");
  return res.data.data;
};

export const updateOrderStatus = async (
  orderId,
  status
) => {
  const res = await api.put(
    `/orders/admin/${orderId}/status`,
    {
      status,
    }
  );

  return res.data.data;
};

export const getAdminOrderById = async (
  orderId
) => {
  const res = await api.get(
    `/orders/admin/${orderId}`
  );

  return res.data.data;
};

export const getSellerOrders = async () => {
  const res = await api.get("/orders/seller/orders");
  return res.data;
};

export const getSellerOrderDetails = async (
  orderItemId
) => {

  const res = await api.get(
    `/orders/seller/orders/${orderItemId}`
  );

  return res.data;
};

export const updateSellerOrderStatus = async (
  orderItemId,
  status
) => {
  const res = await api.put(
    `/orders/seller/orders/${orderItemId}/status`,
    {
      status,
    }
  );

  return res.data;
};

File: backend/src/main/java/com/shop/features/order/service/OrderService.java
package com.shop.features.order.service;

import java.util.List;

import com.shop.features.order.dto.OrderResponseDto;
import com.shop.features.order.dto.PlaceOrderRequestDto;
import com.shop.features.order.dto.UpdateOrderStatusDto;
import com.shop.features.order.entity.UpdateOrderItemStatusDto;
import com.shop.features.seller.dto.SellerOrderResponseDto;

public interface OrderService {

    OrderResponseDto placeOrder(
            String email,
            PlaceOrderRequestDto request
    );

    List<OrderResponseDto> getMyOrders(
            String email
    );

    OrderResponseDto updateOrderStatus(
            Long orderId,
            UpdateOrderStatusDto request
    );

    List<OrderResponseDto> getAllOrders();
    
    OrderResponseDto getOrderById(
    		String email,
    		Long orderId
    		);
    OrderResponseDto getOrderForAdmin(
            Long orderId
    );
    
    List<SellerOrderResponseDto> getSellerOrders(String email);
    				
    SellerOrderResponseDto getSellerOrderDetails(
            String email,
            Long orderItemId
    );
    
    SellerOrderResponseDto updateSellerOrderStatus(
            String email,
            Long orderItemId,
            UpdateOrderItemStatusDto request
    );
    
}